File: Utils/PasswordUtils.fs
//module PasswordUtils

namespace TalentSync.Utils

open System.Security.Cryptography
open System.Text

module PasswordUtils =
    let hashPassword (password: string) =
        use sha = SHA256.Create()
        password
        |> Encoding.UTF8.GetBytes
        |> sha.ComputeHash
        |> Array.map (fun b -> b.ToString("x2"))
        |> String.concat ""

File: Talentsync.CSharp/Utils/PasswordUtils.cs
using System.Security.Cryptography;
using System.Text;

namespace Talentsync.CSharp.Utils
{
    public class PasswordUtils
    {
        public static string hashPassword(string password)
        {
            using (var sha = SHA256.Create())
            {
                var bytes = Encoding.UTF8.GetBytes(password);
                var hash = sha.ComputeHash(bytes);
                var sb = new StringBuilder();
                foreach (var b in hash)
                {
                    sb.Append(b.ToString("x2"));
                }
                return sb.ToString();
            }
        }

        public static bool verifyPassword(string password, string storedHash)
        {
            var hashOfInput = hashPassword(password);
            return hashOfInput == storedHash;
        }
    }
}
